cmd/juju/ssh/ssh_jump.go

```diff
@@ -150,9 +176,28 @@ func (p *sshJump) resolveTarget(target string) (*resolvedTarget, error) {
 	}, nil
 }
 
-// getKeysWithRetry retrieves the public SSH host keys for the target and jump server.
+func (p *sshJump) generateKnownHosts(hostKeys hostKeys) error {
+	knownHosts := newKnownHostsBuilder()
+
+	knownHosts.add(fmt.Sprintf("[%s]:%d", hostKeys.jumpHostname, p.jumpHostPort), []string{hostKeys.jumpHostKey})
+	knownHosts.add(hostKeys.virtualHostname, []string{hostKeys.virtualHostKey})
+
+	f, err := os.CreateTemp("", "ssh_known_hosts")
+	if err != nil {
+		return errors.Annotate(err, "creating known hosts file")
+	}
+	defer func() { _ = f.Close() }()
+	p.knownHostsPath = f.Name() // Record for later deletion
+	if err := knownHosts.write(f); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+// getKeysWithRetry retrieves the public SSH host keys for the target and jump server, and convert them
```

Comment:
```suggestion
// getKeysWithRetry retrieves the public SSH host keys for the jump and target server, and converts them
```
Swap the order to match the order of the keys returned.